distributed/tests/test_scheduler.py

```diff
@@ -1221,6 +1221,27 @@ async def test_profile_metadata(c, s, a, b):
     assert not meta["counts"][-1][1]
 
 
+@gen_cluster(client=True, worker_kwargs={"profile_cycle_interval": 100})
+async def test_profile_metadata_timeout(c, s, a, b):
+    start = time() - 1
+
+    def raise_timeout(*args, **kwargs):
+        raise TimeoutError
+
+    b.handlers["profile_metadata"] = raise_timeout
```

Comment:
Thx. This is something I actually wasn't sure about if this would be "OK". Essentially its monkeypatching which relies on knowledge of the implementation... But I figured the handler concept is pretty stable by now

Reply:
Yeah, it's unfortunate in that if someone changes the implementation they now need to understand this test.  In this case though I'm not sure how else to test this functionality.  Any thoughts?

Reply:
The only thing I can come up with which doesn't require any patching would be to continuously collect profiling information and killing one of the workers while doing so in hopes that the get_profile hits the sweet spot between "worker is dead but scheduler doesn't know about it, yet". I don't really like these timing based things unless absolutely necessary.

I assume the only implementation internal knowledge required here is:
1. There is the concept of a RPC handler
2. This handler is registered using a setitem on the attribute handler
3. The handlers name is profile_metadata

I'd assume 1 is a given and won't change as long as this project exist. Even if it did, I'd expect that these tests would be the least of the problems encountered when trying to change something that fundamental. If 2. or 3. is changed, it is at least not necessary to change or understand the test logic but only to change the assignment here.

TL;DR I believe this is a valid case of patching/mocking, just wanted to bring it up in case somebody else has an issue with this or a better idea